ckanext/canada/templates/internal/package/snippets/stages.html

```diff
@@ -36,7 +36,7 @@
         {# stage 1 #}
         <a><button class="btn-link highlight" name="save" value="go-resources" type="submit">{{ _('Add Resource Metadata') }}</button></a>
       {% else %}
-        {% link_for _('Add data'), controller='package', action='new', class_="highlight" %}
+        {% link_for _('Add data'), 'package.new', class_="highlight" %}
```

Comment:
does this work? shouldn't it be one of the dataset ones?

Reply:
@wardi is does work, but perhaps it is loading the old Controller? There is some stuff in the link_for and url_for that breaks down whether it is a Pylons or Flask thing. I have changed them all to be `dataset` or `pkg.type` now though, as that will set us up for the future. And perhaps make it better for the link_for and url_for to resolve stuff.